lib/scope.js

```diff
@@ -2503,6 +2511,26 @@ module.exports = {
       await scope.detectDownloadComplete( scope );
     },  // Ends scope.steps.download()
 
+    compare_pdfs: async function (scope, {existing_pdf_path, new_pdf_path}) {
+      let full_existing_path = await scope.findFiles(scope, {uploads_names: [existing_pdf_path]});
+      let full_new_pdf_path = `${scope.paths.scenario}/${new_pdf_path}`;
+      let diffs = await pdf_reader.compare_pdfs(full_existing_path[0], full_new_pdf_path);
+      if (diffs) {
+        let diff_str = diffs.reduce((err_str, part) => {
+          if (part.added) {
+            return err_str + `added "${ part.value }", `
```

Comment:
Consider adding newlines?

Test on long PDFs that are completely different.

Consider saving screenshots of the two different PDFs.

Consider adding a line to the log of "If this new PDF is what you want it to be, just replace it". Way in the future, consider a tool that copies the new PDFs into a separate folder, for easy copy-replacing into the sources folder.

Reply:
Follow up on this: New line separated with markdown-like bullets looks good enough IMO. It's not great, but it handles really long PDFs better, and is at least structured a bit more.

![Screenshot from 2023-10-13 14-27-45](https://github.com/SuffolkLITLab/ALKiln/assets/6252212/af419a1d-19d1-438e-b076-73bd03ea226b)

Making an issue for the future work now.

Reply:
PDF fields diff pretty well too, it only shows the different value.

```
ERROR: The PDFs were not the same.
The new PDF added: 
-       "value": "Proxyname1"


The new PDF removed: 
-       "value": "Proxyname12"
```